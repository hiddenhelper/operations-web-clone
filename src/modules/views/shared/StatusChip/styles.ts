import { createStyles, makeStyles } from '@material-ui/core/styles';
import { toREM } from '../../../../utils/generalUtils';
import { STYLE } from '../../../../constants';

export const useStyles = makeStyles(theme =>
  createStyles({
    statusChip: {
      height: '22.5px',
      fontSize: `${toREM(12)}`,
      fontWeight: 600,
      marginLeft: '11px',
      color: STYLE.COLOR.SECONDARY_LIGHTER,
      backgroundColor: STYLE.COLOR.LIGHT_GRAY3,
      cursor: 'inherit',

      '& .MuiChip-label': {
        position: 'relative',
        top: 1,
      },
    },
    tableStatusChip: {
      marginLeft: 0,
    },
    draft: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: STYLE.COLOR.ACCENT_PRIMARY_LIGHTER,
    },
    pendingapproval: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: `${STYLE.COLOR.ACCENT_TERTIARY} !important`,
    },
    assigned: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: STYLE.COLOR.ACCENT_PRIMARY_LIGHTER,
    },
    available: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: STYLE.COLOR.PRIMARY_LIGHTER,
    },
    active: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: `${STYLE.COLOR.PRIMARY_LIGHTER} !important`,
    },
    pending: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: `${STYLE.COLOR.ACCENT_TERTIARY} !important`,
    },
    expired: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: STYLE.COLOR.ERROR_ICON,
    },
    Pending: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: STYLE.COLOR.ACCENT_TERTIARY,
    },
    Accepted: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: STYLE.COLOR.PRIMARY,
    },
    Finished: {
      color: STYLE.COLOR.WHITE,
      backgroundColor: STYLE.COLOR.LIGHT_GRAY8,
    },
  })
);
