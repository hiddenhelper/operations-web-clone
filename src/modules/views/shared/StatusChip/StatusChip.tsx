import React, { memo } from 'react';

import Chip from '@material-ui/core/Chip';

import { useStyles } from './styles';

export interface IStatusChipProps {
  label: string;
  color?: any;
  icon?: React.ReactElement;
  styleClasses?: string;
}

const StatusChip = ({ styleClasses = '', color = 'primary', icon, label }: IStatusChipProps) => {
  const classes = useStyles();
  return <Chip className={`${classes.statusChip} ${styleClasses}`} color={color} label={label} icon={icon} />;
};

export default memo(StatusChip);
