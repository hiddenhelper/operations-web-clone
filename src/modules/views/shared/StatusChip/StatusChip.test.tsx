import React from 'react';
import { render } from '@testing-library/react';

import StatusChip, { IStatusChipProps } from './StatusChip';

describe('StatusChip', () => {
  let props: IStatusChipProps;

  beforeEach(() => {
    props = {
      label: 'test',
    };
  });

  it('should render', () => {
    const { container } = render(<StatusChip {...props} />);
    expect(container).toMatchSnapshot();
  });
});
