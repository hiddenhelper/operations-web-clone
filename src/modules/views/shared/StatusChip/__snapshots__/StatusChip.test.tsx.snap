// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusChip should render 1`] = `
<div>
  <div
    class="MuiChip-root makeStyles-statusChip-1  MuiChip-colorPrimary"
  >
    <span
      class="MuiChip-label"
    >
      test
    </span>
  </div>
</div>
`;
